Doc/library/stdtypes.rst

```diff
@@ -2392,6 +2392,193 @@ that ``'\0'`` is the end of the string.
 .. index::
    single: buffer protocol; binary sequence types
 
+
+.. index::
+   single: formatted string literal
+   single: interpolated string literal
+   single: string; formatted literal
+   single: string; interpolated literal
+   single: f-string
+   single: fstring
+   single: {} (curly brackets); in formatted string literal
+   single: ! (exclamation); in formatted string literal
+   single: : (colon); in formatted string literal
+   single: = (equals); in debug string literal
+.. _f-string-formated-string-literal:
+
+``f-string``-Formatted String Literal
+-------------------------------------
+.. versionadded:: 3.6
+
+A :dfn:`formatted string literal` or :dfn:`f-string` is a string literal that
+is prefixed with ``f`` or ``F``.  These strings may contain replacement
+fields, which are expressions delimited by curly braces ``{}``. While other
+string literals always have a constant value, formatted strings are expressions
+evaluated at run time.::
+
+   >>> f"This is an f-string"
+   'This is an f-string'
+   >>> F"This is also an f-string"
+   'This is also an f-string'
+
+It is also possible to have a multi line string.::
+
+   >>> f'''This is a string
+   ... on two lines'''
+   'This is a string\non two lines'
+
+A single opening curly bracket, ``'{'``, marks a replacement field that contains
+a python expression.::
+
+   >>> name = "Fred"
+   >>> f"He said his name is {name}"
+   'He said his name is Fred'
+
+Everything outside the braces is treated as a literal::
+
+   >>> a = 4
+   >>> b = 3
+   >>> f"The product of {a} and {b} is {a * b}"
+   'The product of 4 and 3 is 12'
+
+If you want to include a literal ``{`` or ``}``, you can double the curly
+brackets to escape them::
+
+   >>> f'{{a}} is {a}'
+   '{a} is 4'
+
+Functions can also be used.::
+
+   >>> s = "Python"
+   >>> f"The string {s.upper()} contains {len(s)} characters."
+   'The string PYTHON contains 6 characters.'
+
+By default the :func:`str` format of a variable is presented when using
+f-strings::
```

Comment:
This isn't technically true, but I don't know if we want to just provide a "you can think of it this way" guide, or go into the real details.

An object's `__format__()` method can do anything. By convention, if there's no format specifier the object's `__format__()` will return `str(obj)`, but it's not a requirement.

Maybe change "By default" to "Typically", or "By convention"?